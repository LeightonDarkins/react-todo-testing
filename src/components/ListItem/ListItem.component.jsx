import React from 'react'
import PropTypes from 'prop-types'

import './ListItem.scss'

const ListItem = ({ itemId, selected, onItemClicked }) => {
  const className = selected ? 'ListItem--selected' : 'ListItem'

  return (
    <li id={itemId}
      data-test-selector='list-item'
      className={className}
      onClick={onItemClicked}>
     Item: {itemId}
    </li>
  )
}

ListItem.propTypes = {
  itemId: PropTypes.string.isRequired,
  selected: PropTypes.bool
}

module.exports = ListItem
