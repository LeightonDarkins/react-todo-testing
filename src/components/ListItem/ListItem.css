.ListItem {
  color: black;
  height: 57px;
  display: flex;
  align-items: center;
  padding-left: 20px;
  border-bottom: 1px solid #223366;
  margin-bottom: 5px;
}

.ListItem:hover {
  color: white;
  cursor: pointer;
  background-color: #223366;
}

.ListItem--selected {
  height: 57px;
  display: flex;
  align-items: center;
  padding-left: 20px;
  border-bottom: 1px solid #223366;
  margin-bottom: 5px;

  color: white;
  background-color: #8899CC;
}
