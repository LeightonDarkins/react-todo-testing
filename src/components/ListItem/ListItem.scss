.ListItem {
  color: black;
  height: 57px;
  display: flex;
  align-items: center;
  padding-left: 20px;
  border-bottom: 1px solid #223366;
  margin-bottom: 5px;

  &:hover {
    color: white;
    cursor: pointer;
    background-color: #223366;
  }

  &--selected {
    @extend .ListItem;

    color: white;
    background-color: #8899CC;
  }
}
