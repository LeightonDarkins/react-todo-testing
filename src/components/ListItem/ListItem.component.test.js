/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import ListItem from './ListItem.component.jsx'

let wrapper

describe('ListItem', () => {
  afterEach(() => {
    wrapper = undefined
  })

  it('renders a selected list item', () => {
    const tree = renderer
      .create(<ListItem itemId='test-id' selected />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders a regular list item', () => {
    const tree = renderer
      .create(<ListItem itemId='test-id' />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('propagates an event when click', () => {
    const mockFunction = jest.fn()

    wrapper = shallow(<ListItem itemId='test-id' onItemClicked={mockFunction} />)

    wrapper.simulate('click')

    expect(mockFunction).toHaveBeenCalledTimes(1)
  })
})
