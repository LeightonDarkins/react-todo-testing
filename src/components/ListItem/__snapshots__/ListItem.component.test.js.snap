// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem renders a regular list item 1`] = `
<li
  className="ListItem"
  data-test-selector="list-item"
  id="test-id"
  onClick={undefined}
>
  Item: 
  test-id
</li>
`;

exports[`ListItem renders a selected list item 1`] = `
<li
  className="ListItem--selected"
  data-test-selector="list-item"
  id="test-id"
  onClick={undefined}
>
  Item: 
  test-id
</li>
`;
